x/exchange/keeper/commitments.go

```diff
@@ -426,3 +457,43 @@ func (k Keeper) consumeCommitmentSettlementFee(ctx sdk.Context, req *exchange.Ms
 	antewrapper.ConsumeMsgFee(ctx, exchangeFees.ExchangeFees, req, "")
 	return nil
 }
+
+// validateTransferCommitments which validate TransferCommitments message request.
+func (k Keeper) validateTransferCommitments(
+	ctx sdk.Context,
+	store storetypes.KVStore,
+	req *exchange.MsgMarketTransferCommitmentsRequest,
+	account sdk.AccAddress,
+) error {
+	// Blocked address check
+	if k.bankKeeper.BlockedAddr(account) {
+		return fmt.Errorf("%s is not allowed to receive funds", account)
+	}
+	// Zero amount
+	if req.Amount.IsZero() {
+		return fmt.Errorf("account %s does not have any funds committed to market %d", account, req.CurrentMarketId)
+	}
```

Comment:
```suggestion
	if req.Amount.IsZero() {
		return fmt.Errorf("cannot transfer zero for %s in market %d", account, req.CurrentMarketId)
	}
```

The `req.Amount` is how much they're asking to transfer, not how much they actually have committed.